tests/unit/test_chat_references.py

```
@@ -637,3 +637,110 @@ async def test_ask_deduplicates_references(self, auth_tokens, httpx_mock):
+        for chunk in chunks:
+            inner_json = json.dumps(chunk)
+            chunk_json = json.dumps([["wrb.fr", None, inner_json]])
+            parts.append(f"\n{len(chunk_json)}\n{chunk_json}")
+        parts.append("\n")
+        return "".join(parts)
+
+    def test_multi_chunk_does_not_inflate_references(self, chat_api):
+        """References must not multiply when the API streams repeated chunks.
+
+        Issue #300: each streaming chunk repeated the full reference list, so
+        N chunks × M refs = N*M entries instead of M. The fix tracks refs
```

Ruff (0.15.11)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)